### ISSUE TITLE
Add check for list comprehensions in any or all calls

### ISSUE THREAD

**Issue by tweakimp** [astral-sh/ruff#19330]
### Summary

Problem:
There are many cases of comprehensions inside of `any` or `all` calls that are unnecessary and should be replaced by a generator.

For example:
```python
some_list = list(range(1000))
test_old = any([x % 7 == 0 for x in some_list])
````
Instead, a generator would be less code, and way faster:
```python
test_new = any(x % 7 == 0 for x in some_list)
```

Speed comparisons:
```
64 µs ± 1.67 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each) # old
447 ns ± 3.48 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each) # new
```

`any` and `all` are functions that profit from that because the loop is breaking and can short cut; the remaining elements are not even looped over if the test returns a `True` in the `any` function or a `False` in the `all` function.

**Comment by dylwil3** [comment]
Thanks for the suggestion! Do you have an example that isn't captured by [unnecessary-comprehension-in-call (C419)](https://docs.astral.sh/ruff/rules/unnecessary-comprehension-in-call/#unnecessary-comprehension-in-call-c419)? The one in your post seems to be: [Playground link](https://play.ruff.rs/80d8426a-2112-48e3-9a20-d1acc6dfc222)

**Comment by tweakimp** [comment]
Ah my bad, I didnt have all rules enabled in the default setting when just running `ruff check .`
